solcore/optics/external_optics.py

```diff
@@ -41,6 +88,47 @@ def solve_external_optics(solar_cell, options):
     solar_cell.absorbed = all_absorbed * initial
 
 
-def absorbed(self, z):
-    out = self.diff_absorption(self.offset + z) * (z < self.width)
-    return out.T
+def solve_external_optics_matrix(solar_cell, options):
+    """ Uses the reflection, transmission and absorption of a solar cell from
+    external data, preparing the structure for further calculations. The
+    external data that must be provided in the solar cell class as attributes
+    are:
+
+    RAT: a dictionary containing  the R, T, A numpy arrays {'R':R, 'T':T, 'A':A}
+    absorbed_matrix: a matrix containing the absorption as A[wl,z]. 
+
+    Note that both are based on arrays, therefore the user should provide them with
+    the same spectral, and z spacing as the simulation indicated in the options.
+
+    :param solar_cell: 
+    :param options: 
+    :return:
+    """
+    wl = options.wavelength
+    # We include the shadowing losses
+    initial = (1 - solar_cell.shading) if hasattr(solar_cell, 'shading') else 1
+    try:
+        RAT = solar_cell.RAT 
+        absorbed_matrix = solar_cell.absorbed_matrix 
+    except AttributeError as err:
+        raise err
+
+    position = options.position * 1e9
+    delta = (position[1] - position[0])
```

Comment:
This will not work well if the positions are inhomogeneous, eg. denser points near the surface and then much more spaced ones in the thick layers. I guess it can be easily reformulated using [numpy.diff](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.diff.html) or something similar.

Reply:
Ok, then we have to rescale the aborption_power, maybe something like this?
```python
# delta = (position[1] - position[0])
 delta = 1./np.diff((position)
 absorption_power = absorbed_matrix[:,1:] * (delta)
```